tensornetwork/network_operations.py

```diff
@@ -309,6 +317,12 @@ def split_node(
     edge.update_axis(i + len(left_edges), node, i + 1, right_node)
   connect(left_node.edges[-1], right_node.edges[0], name=edge_name)
   node.fresh_edges(node.axis_names)
+  #rerstore the original edge order of node
+  orig_edge_order = [None] * node.get_rank()
+  for n, o in enumerate(orig_axes):
+    orig_edge_order[o] = node.edges[n]
+  node.reorder_edges(orig_edge_order)
+
```

Comment:
I'm a little against this solution. It does a second unnecessary transpose which could cause unintended slow downs, especially when running in a distributed setting. 

I think a better approach would be to add a `node.tensor_from_edge_order(edges)` which would return the transposed tensor but not modify `node`.

Reply:
good point. why not `Node.transpose(List[Union[int,Edge]])`?

Reply:
never mind, I think `tensor_from_edge_order` is better.